### PR TITLE
Fix #2473: prune retired nodes on a full node-repo install

### LINKED ISSUE #2473: A node the source repo retires survives in the mesh — and a retired NodeType parks the portal on the next framework flip
Split out of #1786, whose two remaining incidents both trace to this and not to the content.

## The condition

**A node the source repo deletes stays in the mesh forever.** It is inert until the framework identity moves — at which point, if it is a NodeType, it is recompiled with zero sources, fails, and **parks**. A parked NodeType refuses portal readiness and holds every instance hub for the full 60 s activation budget (AGENTS.md).

Measured live on `memex.meshweaver.cloud` today (`3.0.0-rc8.ci.6011`), the whole mesh has exactly **one** NodeType at `CompileError` and this is it:

```
LinkedIn/Skill
CS0246 The type or namespace name 'Skill' could not be found
--- Source discovery ---
  - namespace:LinkedIn/Skill/Source scope:subtree nodeType:Code
Matched Code nodes (0):
  (none) — the configuration lambda cannot reference types because no source files were included.
```

`MeshWeaver.SocialMedia@75f3f3d` (2026-08-20) deliberately retired that type — it shadowed the platform's reserved `Skill` name, nothing produced a single instance, and its one reader was moved to `LinkedIn/LinkedInSkill` in the same commit. The repo dropped `LinkedIn/Skill.json` **and** `LinkedIn/Skill/Source/Skill.cs`. The mesh kept the NodeType, with an empty `Source/`.

## Why it only detonated a week later

The node's own record:

```
lastCompileSucceededAt    2026-08-21T13:38:24Z   (assembly LinkedIn_Skill/v848-sbf2f93e-….dll)
compiledFrameworkVersion  sbf2f93e099b84829478731f0c64835b9
lastCompileStartedAt      2026-08-27T06:16:17Z
failedBuildInputs         fw=s990240a13187861c03d0a2db1cb67775; …; src=0
currentSourceVersions     {}
```

It served its already-built assembly until the framework identity moved. A bump recompiles **every** dynamic NodeType, so the orphan's brokenness arrives as an event, not a trickle — and it arrives at the worst moment, during a roll. This compounds with MeshWeaver.Plugins#723.

## Why nothing removed it

**Only the DELTA install prunes.** `PackageInstaller.InstallNodeRepoDelta` deletes `removedNodePaths` derived from a manifest diff. The FULL install has no prune at all, deliberately — `PackageInstaller.cs:25` explains that full-replace semantics would wipe the rest of a shared partition. The delta is unavailable whenever:

* there is no install record carrying `InstalledFiles` to diff against (`CatalogLayoutAreas.cs:593` — `newManifest.DiffFrom(record.InstalledFiles)`), or
* the diff touches the package's shared `Source/` / `Test/`, which throws *"changed shared Source/Test files; full install required"* (`CatalogLayoutAreas.cs:598`).

And even on the delta path the prune is **silent when it fails**:

```csharp
.Catch<int, Exception>(ex =>
{
    logger?.LogWarning(ex, "Pruning removed node {Path} failed.", path);
    return Observable.Return(0);
})
```

so the install's `{Pruned} pruned` line under-reports and nothing downstream — CI, the gate, the portal — ever notices that an orphan survived.

## What would actually close it

Not "delete harder". Three candidates, in increasing order of ambition:

1. **Make the survivor visible.** A parked NodeType whose source set is EMPTY is a distinct condition from "the code is broken" — `src=0` with a configuration lambda naming a type is always an orphan or a mis-pointed `sources` list, never a code error. Report it as such (its own log line / an entry in the readiness report), so it is one grep away instead of one hand-sweep away.
2. **Give the full install a bounded prune.** The objection at `PackageInstaller.cs:25` is about wiping a *shared* partition, which is a scoping problem, not an impossibility: the installer knows the package's own file set and its install record's previously-installed paths. Pruning exactly `previously-installed ∖ now-shipped` is the delta's rule applied without a diff.
3. **Stop swallowing the prune failure.** A failed prune is not a cosmetic loss — it is precisely how this incident happened. At minimum it must not be counted as `0`; ideally an install that could not remove a node it was asked to remove is not a successful install.

## Also still open from #1786 (unchanged, restated so it is not lost)

**A NodeType that has NEVER compiled successfully and settles at `Error` is unreachable by every automatic re-drive**, so a fix to the cause can never reach the node: `ApplyCompileFailure` writes no `CompiledFrameworkVersion`, and `HasStaleFrameworkBuild` requires the assembly coordinates it never writes. The framework-stale kickoff's own `Where` already reads `CompilationStatus is Ok or Error`, so covering `Error` is clearly the intent. `LinkedIn/Skill` escaped that trap only because it had *previously* succeeded.

## Immediate cleanup this issue does not cover

`LinkedIn/Skill` needs deleting on memex-cloud. MCP refuses it for `rbuergi` (no Delete on its `_Activity` satellite) — someone with Delete on the LinkedIn partition can do it at <https://memex.meshweaver.cloud/LinkedIn/Skill/Delete>. `NodeTypeUnparkPostDeletionHandler` clears the park on delete.

### PR DESCRIPTION
**Pull request by rbuergi**
## Summary

Closes #2473.

A node the source repo deletes used to stay in the mesh forever whenever the
plugin update fell back to a **full** node-repo install instead of the
manifest-diff **delta** path — the only path that pruned removed nodes. The
fallback is not an edge case: `CatalogLayoutAreas.IncrementalUpdate` refuses
and falls back to full whenever the update touches the package's shared
`Source/`/`Test/` (whose blast radius is every type in the package), or
whenever there is no previous install record with a file-level baseline.

Concretely, this is how `LinkedIn/Skill` (MeshWeaver.SocialMedia@75f3f3d)
survived in `memex.meshweaver.cloud` after its file was deleted from the
source repo: it kept serving its last-built assembly with an empty
`Source/` until the next framework-identity flip forced every dynamic
NodeType to recompile — at which point it failed against zero sources,
parked at `CompileError`, and held every instance hub for the full 60s
activation budget (see `NodeTypeCompilation.md`). This is a live,
currently-present condition on memex-cloud today, not a hypothetical: the
NodeType is still the only one on that mesh sitting at `CompileError`, with
`src=0` source discovery and its entire `Release/*` compile-history subtree
still attached under the retired type's path.

## Fix

`PackageInstaller.InstallNodeRepo` (the full-install path) now:
- reads the package's own previous install record (`{Plugins}/{Id}`) for
  its `InstalledFiles` baseline, before that record is overwritten;
- diffs it against the incoming `manifest.lock` via the same
  `ModuleManifest.DiffFrom` the delta path already uses;
- prunes exactly `previously-installed ∖ now-shipped`, through the same
  user-claim fence (`SyncBehavior.Include`) and System-impersonated delete
  the delta path uses — via `IMeshService.DeleteNode`, which is always
  **recursive** (`MeshService.DeleteNode` hard-codes `Recursive = true`),
  so pruning a retired NodeType takes its entire subtree with it in one
  call — including compile-history `Release/*` children, which are never
  routed to a satellite table (`PartitionDefinition.StandardTableMappings`
  never routed `_Release`/`Release` — see Memex.Database.Migration's
  `V19_DeleteLegacyReleaseNodes`) and so live as ordinary `mesh_nodes` rows
  squarely inside the deleted path prefix. Verified with a dedicated test
  (see Red-proof) rather than assumed from reading the delete path.

Bounded by construction: the candidate set only ever contains paths *this
package's own* previous install record listed, so it can never touch a
shared partition's other content — the same reasoning that already makes
the delta path's prune safe, applied without a caller-handed diff. The
delta path's prune logic is now a shared helper (`PruneRemovedNodes`)
instead of duplicated.

No-ops safely when there's nothing to prune: a fresh install (no previous
record) or a package that ships no `manifest.lock` (no file-level baseline
exists at all) — matching the two conditions the issue names as the ones
that make the delta path unavailable.

**Review fix**: the automatic Copilot review flagged that the shared prune
helper used `Observable.Using(() => access.ImpersonateAsSystem(), …)` —
the exact shape #1790 banned (impersonation is an `AsyncLocal` store/restore
pair; `Using` opens on the subscribing thread but disposes on the
*terminating* thread, which for a cross-hub delete latches `system-security`
onto whatever runs next on the subscriber). Replaced with the sealed
`AccessService.RunAsSystem` helper, which opens and closes the scope inside
one synchronous `Subscribe`. This also retroactively fixes the identical
pre-existing pattern in the delta path's `Prune()`, which now shares this
helper.

## Red-proof

- `RetiredNodePruneTest.SharedSourceChange_FallsBackToFullInstall_AndPrunesTheRetiredNodeType`
  reproduces the exact trigger: a NodeType (`Widget/Thing`) is retired in the
  same update that also changes the package's shared `Source/`, which forces
  `IncrementalUpdate` to throw and `InstallOrUpdateCore` to fall back to
  `Full()`. Confirmed RED against the pre-fix code (reverted locally, times
  out waiting for the retired node to disappear — `TimeoutException` at 31s)
  and GREEN with the fix.
- `RetiredNodePruneTest.ReleaseHistorySurvivesUnderThePrunedType_ThenIsPrunedWithIt`
  plants a `Release`-shaped child directly under the doomed type (the way a
  live compile would, never through the package installer, since compile
  history is never a repo file and so never appears in any manifest diff),
  then asserts it disappears along with its owning type — empirically
  confirming the prune is a subtree delete, not a single-node delete, which
  is what protects against the orphaned-`Release`-history shape found live
  on memex-cloud.
- Full `MeshWeaver.PluginCatalog.Test` suite (651 tests, including
  `IncrementalUpdateTest` and `NodeRepoInstallTest`, which exercise the code
  this change refactors) — all green, no regressions.

## What's New

`src/MeshWeaver.Documentation/Data/WhatsNew/2026-08-28-retired-node-prune.md`
— `Category: Fix` (a user/operator can notice a plugin update leaving stale
content behind, and the downstream portal-readiness stall it can cause).

## Test plan

- [x] `dotnet build src/MeshWeaver.PluginCatalog -c Release -warnaserror` — clean
- [x] `dotnet build test/MeshWeaver.PluginCatalog.Test -c Release -warnaserror` — clean
- [x] `RetiredNodePruneTest` (both cases) — RED on reverted code, GREEN with the fix
- [x] Full `MeshWeaver.PluginCatalog.Test` suite — 651/651 passed
- [x] Automatic Copilot review addressed (`RunAsSystem`) and thread resolved
- [ ] CI green (`Consolidate test results`)

## Not covered by this PR

`LinkedIn/Skill` itself still needs deleting on memex-cloud — this PR fixes
the mechanism so a FUTURE retirement prunes correctly; it does not reach
back and clean up a node that already survived past the point this fix
lands. That cleanup needs someone with Delete on the LinkedIn partition
(noted in #2473's own "Immediate cleanup this issue does not cover"
section).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/MeshWeaver.PluginCatalog/PackageInstaller.cs [hunk 1]
```diff
@@ -2675,6 +2686,107 @@ placeholderRoot is not null && root is not null
...
+                    : Observable.Return<MeshNode?>(null))
+                .SelectMany(current =>
+                    current is not null && current.SyncBehavior != SyncBehavior.Include
+                        ? Observable.Return(0)
+                        : Observable.Using(
+                                () => accessService?.ImpersonateAsSystem()
+                                      ?? System.Reactive.Disposables.Disposable.Empty,
+                                _ => meshService.DeleteNode(path))
+                            .Take(1)
+                            .Select(deleted => deleted ? 1 : 0))
```
[the remark this answers is not in the document]
Comment (rbuergi) [reply src/MeshWeaver.PluginCatalog/PackageInstaller.cs]: Confirmed and fixed in 3697ef14a: replaced `Observable.Using(() => accessService?.ImpersonateAsSystem() ?? Disposable.Empty, _ => meshService.DeleteNode(path))` with `accessService.RunAsSystem(() => meshService.DeleteNode(path))` in the new shared `PruneRemovedNodes` helper. You're right that Using's dispose lands on the terminating thread (the owning hub's response thread for a cross-hub delete), not the subscribing thread that opened the scope — RunAsSystem seals both open and close inside the one Subscribe (per ImpersonationScopeExtensions/#1790), so this can't latch system-security onto whatever runs next. This also retroactively fixes the same pre-existing pattern in the delta path's Prune(), which now shares this helper. Rebuilt with -c Release -warnaserror (clean) and reran the full MeshWeaver.PluginCatalog.Test suite (650/650 green).
